fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java

```diff
@@ -4389,7 +4389,7 @@ public Map<String, String> getForwardVariables() {
             Field[] fields = SessionVariable.class.getDeclaredFields();
             for (Field f : fields) {
                 VarAttr varAttr = f.getAnnotation(VarAttr.class);
-                if (varAttr == null || !varAttr.needForward()) {
+                if (varAttr == null || !varAttr.needForward() || !varAttr.affectQueryResult()) {
```

Comment:
i think this is wrong, should be
```suggestion
                if (varAttr == null || !(varAttr.needForward() || varAttr.affectQueryResult())) {
```